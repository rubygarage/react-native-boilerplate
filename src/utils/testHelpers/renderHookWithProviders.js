/* eslint-disable import/no-extraneous-dependencies */
import { renderHook } from '@testing-library/react-hooks';
import { ThemeProvider } from 'styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import mockedTheme from 'utils/testHelpers/mockedTheme';

const store = configureStore()({});
store.dispatch = jest.fn();

const renderHookWithProviders = (hook) => {
  const wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={mockedTheme}>
        {children}
      </ThemeProvider>
    </Provider>
  );

  return renderHook(hook, { wrapper });
};

export default renderHookWithProviders;
