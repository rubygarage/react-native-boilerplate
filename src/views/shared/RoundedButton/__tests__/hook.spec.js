import renderHookWithProviders from 'utils/testHelpers/renderHookWithProviders';

import useContainer from '../hook';

describe('RoundedButton useContainer hook', () => {
  it('matches snapshot', () => {
    const { result } = renderHookWithProviders(useContainer);

    expect(result.current).toMatchSnapshot();
  });
});
