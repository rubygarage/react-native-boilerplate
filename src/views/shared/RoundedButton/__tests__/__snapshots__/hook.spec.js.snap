// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton useContainer hook matches snapshot 1`] = `
Object {
  "theme": Object {
    "colors": null,
    "fonts": null,
    "sizes": null,
    "type": "Mocked theme",
  },
}
`;
